### PR TITLE
Add access to internal representation of geo_point doc values

### PR DESCRIPTION
**Pull request by iverase**
While working in adding support on top metrics for geo_point field type, I notice it would be grate if we could access the encoded / internal representation of a geo_point. While doing so, I notice as well that there were some bits on the doc values plumbing which were not used at all, for example we were never creating instances of `SingletonMultiGeoPointValues`.

Another thing to notice is that `MultiGeoPointValues` is just a thin wrapper on top of `SortedNumericDocValues` and `GeoPointValues` is a wrapper on top of `NumericDocValues`.

This PR makes those thin wrappers final classes and expose the internal  `SortedNumericDocValues` for geo_points. This simplifies the code and gives the chance to aggregations to follow more efficient paths. In addition this PR implements properly the singleton logic for geo_point doc values.

The only downside of this approach is that we change the behaviour of missing values and with this new approach values get quantised while until now the missing values were not changed. I think this approach is better as the current missing values can be values that do not exist in the index which it seems to me like a bug.

### CODE REVIEW

**Review by not-napoleon — APPROVED** [review]
I think this is a good change.  I don't have enough perspective to judge the missing values part though.  It doesn't seem like a bug to me that missing values can be something not in the index; that's normal for other field types, unless I'm misunderstanding what you mean.  Is there any chance this could be considered a breaking change?  Other than that, LGTM.

File: server/src/main/java/org/elasticsearch/common/geo/GeoPoint.java [hunk 1]
```diff
@@ -197,6 +197,16 @@ public String getGeohash() {
         return Geohash.stringEncode(lon, lat);
     }
 
+    public long getEncoded() {
```
[the remark this answers is not in the document]
Comment (iverase) [reply server/src/main/java/org/elasticsearch/common/geo/GeoPoint.java:200]: Added

**Comment by iverase** [comment]
>  It doesn't seem like a bug to me that missing values can be something not in the index
Mi idea is that adding a missing value should be the same as if we have the same index and documents without value are now indexed with the default value. The result from those two indexes should be the same. In this case that is not true because the value returned at the moment by the default value does not exist in the numerical space of the index

>  Is there any chance this could be considered a breaking change?
I don't think so in the case that we are just returning the default value under the precision accuracy that we claim the index have.

**Comment by not-napoleon** [comment]
Sorry I didn't get back to you yesterday.  I think your explanation makes sense, thanks.

**Pull request merged**